app/controllers/user.js

```diff
@@ -39,6 +43,40 @@ exports.create = (req, res) => {
     });
 };
 
+exports.generate = (req, res) => {
+  const username = short.generate();
+  const cardKey = short.generate();
+  const password = crypto.randomBytes(10).toString('hex');
+  const unActivatedEmail = req.body.email;
+  const userObject = {
+    username,
+    password,
+    cardKey,
+    unActivatedEmail,
+  };
+  const user = new User(userObject);
+  return User.register(user, password)
+    .then((createdUser) =>
+      mailHandler(userObject)
+        .then(() => createdUser.getCleanUser())
+        .then((user) => res.status(201).json(user))
```

Comment:
We do probably not need this in the frontend?

Reply:
very true, initially i didn't plan for the email implementation. So I'll remove this